src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -108,20 +110,48 @@ export class CommandLineService implements ICommandLineProcessing {
 			}
 			await this._commandService.executeCommand(commandName, connectedContext);
 		} else if (profile) {
-			if (this._statusBarService) {
-				this._statusBarService.setStatusMessage(localize('openingNewQueryLabel', 'Opening new query:') + profile.serverName, 2500);
+			// If we were given a file and it was opened with the sql editor,
+			// we want to connect the given profile to to it.
+			// If more than one file was passed, only show the connection dialog error on one of them.
+			if (args._ && args._.length > 0) {
+				await args._.forEach((f, i) => this.processFile(URI.file(f).toString(), profile, i === 0));
```

Comment:
What happens if we're passed invalid file paths (such as arguments that aren't paths)?

Reply:
then there won't be any open editors that match

Reply:
Interesting - if you pass "X:\foo" on the command line, ADS opens an editor named "foo"
The uri doesn't match what's passed in so we don't connect it. Since this change is meant to support ssmsmin launching ADS with a known good file path I don't consider this a blocking issue.

Reply:
I'm saying for the URI.file(f) call - what's the behavior for non-path strings? Brief glance at URI.file seems like it'll be fine but make sure you double check that since I'd expect that to be a valid scenario

Reply:
all that does is create a uri like "file://<bogus uri>" so it never matches anything
I've tried a bunch of random stuff like "https://foo.com/myfile"

ADS creates the editor but it doesn't connect.

Though trying to close ADS I see an error "Cannot ready property 'message' of undefined"

I need to see if that's related.

Reply:
now I can't repro it